# Hi, Welcome to this Data Analysis project.
## I'm going to uncover an exploratory data analysis consisting in the 'churn rate' of a bank. We are going to ask ourselves some general questions, do some comparations / relations and gatter some interesting data.

First, what is "churn rate? churn rate means "the annual percentage rate at which customers stop subscribing to a service or employees leave a job."

Dataset is in the github file:


Dataset name: Churn_Modelling.csv


We start by charging the libraries we are going to use

In [9]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

We charge the dataset via csv

In [10]:
data = pd.read_csv('Churn_Modelling.csv')

### We use info to take a look to the data, to take insights and start the analysis.

In [11]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   RowNumber        10000 non-null  int64  
 1   CustomerId       10000 non-null  int64  
 2   Surname          10000 non-null  object 
 3   CreditScore      10000 non-null  int64  
 4   Geography        10000 non-null  object 
 5   Gender           10000 non-null  object 
 6   Age              10000 non-null  int64  
 7   Tenure           10000 non-null  int64  
 8   Balance          10000 non-null  float64
 9   NumOfProducts    10000 non-null  int64  
 10  HasCrCard        10000 non-null  int64  
 11  IsActiveMember   10000 non-null  int64  
 12  EstimatedSalary  10000 non-null  float64
 13  Exited           10000 non-null  int64  
dtypes: float64(2), int64(9), object(3)
memory usage: 1.1+ MB


In [12]:
df = pd.DataFrame(data)
print(df)

      RowNumber  CustomerId    Surname  CreditScore Geography  Gender  Age  \
0             1    15634602   Hargrave          619    France  Female   42   
1             2    15647311       Hill          608     Spain  Female   41   
2             3    15619304       Onio          502    France  Female   42   
3             4    15701354       Boni          699    France  Female   39   
4             5    15737888   Mitchell          850     Spain  Female   43   
...         ...         ...        ...          ...       ...     ...  ...   
9995       9996    15606229   Obijiaku          771    France    Male   39   
9996       9997    15569892  Johnstone          516    France    Male   35   
9997       9998    15584532        Liu          709    France  Female   36   
9998       9999    15682355  Sabbatini          772   Germany    Male   42   
9999      10000    15628319     Walker          792    France  Female   28   

      Tenure    Balance  NumOfProducts  HasCrCard  IsActiveMemb

## We start the cleaning phase, to start the analysis
### We are going to drop some irrelevant data and clean duplicates (If there's some)

In [13]:
data = data.drop(columns=['RowNumber', 'CustomerId', 'Surname',])
print(data)

      CreditScore Geography  Gender  Age  Tenure    Balance  NumOfProducts  \
0             619    France  Female   42       2       0.00              1   
1             608     Spain  Female   41       1   83807.86              1   
2             502    France  Female   42       8  159660.80              3   
3             699    France  Female   39       1       0.00              2   
4             850     Spain  Female   43       2  125510.82              1   
...           ...       ...     ...  ...     ...        ...            ...   
9995          771    France    Male   39       5       0.00              2   
9996          516    France    Male   35      10   57369.61              1   
9997          709    France  Female   36       7       0.00              1   
9998          772   Germany    Male   42       3   75075.31              2   
9999          792    France  Female   28       4  130142.79              1   

      HasCrCard  IsActiveMember  EstimatedSalary  Exited  
0   

# In this part of the **analysis (EDA)** we are going to uncover and answer some questions about the data.

## There's **10,000 people** in this dataset.
## There's **5,151 active** members and **4,849** Innactive.
### This means **4849** people have officially 'churn'.
#### These people are from three different countries: **Spain, Germany and France.**
##### **France:** 5,014
##### **Germany**: 2.509
##### **Spain**: 2,477
#### There's **5,457 Male's** and **4,543 Female's**.
#### The minimal age is **18** and the max is **92**.
#### The median age is **38**.


##### **All the cells related to this info are below**.

In [14]:
active_members = data['IsActiveMember'].value_counts().apply(lambda x: f'{x:,.0f}')
print(active_members)

IsActiveMember
1    5,151
0    4,849
Name: count, dtype: object


In [15]:
gender_distribution = data['Gender'].value_counts().apply(lambda x: f'{x:,.0f}')
print(gender_distribution)

Gender
Male      5,457
Female    4,543
Name: count, dtype: object


In [16]:
gender_distribution = data['Geography'].value_counts().apply(lambda x: f'{x:,.0f}')
print(gender_distribution)

Geography
France     5,014
Germany    2,509
Spain      2,477
Name: count, dtype: object


In [17]:
min_age = data['Age'].min()
max_age = data['Age'].max()
median_age = data['Age'].mean()
median_age = int(median_age)

print(f"Min age is: {min_age}")
print(f"Max age is: {max_age}")
print(f"Median age is: {median_age}")

Min age is: 18
Max age is: 92
Median age is: 38


## In this point of the **analisis**, we are going to focus just on the people that have **'churn'**. The people that are no longer associated to the bank.


Churned amount of people by country:

*  France  -   2,591
*  Spain   -  1,312
*  Germany -  1,248

Churned people by gender:

*  Male: - 2, 590
*  Female: - 2, 259

Mean age of people churned: 37 years

Avg age by gender:

*  Females: 38 years
*  Males: 37 years

Mean salary of customers(annual): 99,452

Mean salary per country:

*  France   -   98,887
*  Germany  - 100,577
*  Spain   -    99,503

Credit score of churned customers:

*  Mean: - 652
*  Max: - 850
*  Min: - 350

**All the cells related to this info are below**.

In [38]:
churn_rate = data[data['IsActiveMember'] == 1]
print(churn_rate)

      CreditScore Geography  Gender  Age  Tenure    Balance  NumOfProducts  \
0             619    France  Female   42       2       0.00              1   
1             608     Spain  Female   41       1   83807.86              1   
4             850     Spain  Female   43       2  125510.82              1   
6             822    France    Male   50       7       0.00              2   
8             501    France    Male   44       4  142051.07              2   
...           ...       ...     ...  ...     ...        ...            ...   
9986          673   Germany    Male   47       1  183579.54              2   
9987          606     Spain    Male   30       8  180307.73              2   
9989          841     Spain    Male   28       4       0.00              2   
9996          516    France    Male   35      10   57369.61              1   
9997          709    France  Female   36       7       0.00              1   

      HasCrCard  IsActiveMember  EstimatedSalary  Exited  
0   

Churned amount of people by country

In [35]:
churn_by_country = churn_rate['Geography'].value_counts().apply(lambda x: f'{x:,.0f}')
print(churn_by_country)

Geography
France     2,591
Spain      1,312
Germany    1,248
Name: count, dtype: object


Churned people in total filtered by gender.
Male: 2, 590
Female: 2, 259

In [24]:
churn_gender_distribution = churn_rate['Gender'].value_counts().apply(lambda x: f'{x:,.0f}')
print(churn_gender_distribution)

Gender
Male      2,867
Female    2,284
Name: count, dtype: object


Mean age of people churned: **37**

1.   Females: 38
2.   Males: 37



In [25]:
churn_age_distribution = churn_rate['Age'].value_counts().sort_index()
churn_age_mean = churn_rate['Age'].mean()
churn_age_mean = int(churn_age_mean)

churn_age_male = churn_rate[churn_rate['Gender'] == 'Male']
churn_age_female = churn_rate[churn_rate['Gender'] == 'Female']
churn_age_male_mean = churn_age_male['Age'].mean()
churn_age_female_mean = churn_age_female['Age'].mean()
churn_age_male_mean = int(churn_age_male_mean)
churn_age_female_mean = int(churn_age_female_mean)

print(f"Mean age of churned customers: {churn_age_mean} years")
print(f"Mean age of churned male customers: {churn_age_male_mean} years")
print(f"Mean age of churned female customers: {churn_age_female_mean} years")

Mean age of churned customers: 39 years
Mean age of churned male customers: 39 years
Mean age of churned female customers: 39 years


In [46]:
churn_rate_salary = churn_rate['EstimatedSalary'].mean()
churn_rate_salary = int(churn_rate_salary)

print(f"Mean salary of churned customers: {churn_rate_salary}")

Mean salary of churned customers: 99452


Mean salary by churned customers by country

In [27]:
churn_rate_salary_mean_by_country = churn_rate.groupby('Geography')['EstimatedSalary'].mean()
# Format the mean estimated salary by country with commas
churn_rate_salary_mean_by_country_formatted = churn_rate_salary_mean_by_country.apply(lambda x: f'{x:,.0f}')

print("Mean estimated salary of churned customers by country (formatted):")
print(churn_rate_salary_mean_by_country_formatted)

Mean estimated salary of churned customers by country (formatted):
Geography
France      98,887
Germany    100,577
Spain       99,503
Name: EstimatedSalary, dtype: object


Credit score Mean, max and min

In [34]:
churn_score_mean = churn_rate['CreditScore'].mean()
churn_score_min = churn_rate['CreditScore'].min()

print(f"Mean credit score of churned customers: {churn_score_mean}")
print(f"Max credit score of churned customers: {churn_score_max}")
print(f"Min credit score of churned customers: {churn_score_min}")

Mean credit score of churned customers: 652.9341875364007
Max credit score of churned customers: 850
Min credit score of churned customers: 350


# Final report

There's 10,000 people in this dataset.


There's 5,151 active members and 4,849 Innactive.


This means 4849 people have officially 'churn'.


These people are from three different countries: Spain, Germany and France.

*  France: 5,014
*  Germany: 2.509
*  Spain: 2,477


There's 5,457 Male's and 4,543 Female's.


The minimal age is 18 and the max is 92.


The median age is 38.


Churned amount of people by country:


*  France - 2,591
*  Spain - 1,312
*  Germany - 1,248


Churned people by gender:


*  Male: - 2, 590
*  Female: - 2, 259


Mean age of people churned: 37 years

Avg age by gender:


*  Females: 38 years
*  Males: 37 years


Mean salary of customers(annual): 99,452


Mean salary per country:

*  France - 98,887
*  Germany - 100,577
*  Spain - 99,503


Credit score of churned customers:

*  Mean: - 652
*  Max: - 850
*  Min: - 350